pkgs/applications/misc/collision/default.nix

```diff
@@ -0,0 +1,107 @@
+{ lib
+, fetchFromGitHub
+, crystal
+, libadwaita
+, gtk3
+, gobject-introspection
+, openssl
+, libxml2
+, pkg-config
+, wrapGAppsHook4
+}:
+let
+  gi_crystal = crystal.buildCrystalPackage {
+    name = "gi-crystal";
+
+    src = fetchFromGitHub {
+      owner = "hugopl";
+      repo = "gi-crystal";
+      rev = "refs/tags/v0.16.0";
```

Comment:
```suggestion
    pname = "gi-crystal";
    version = "0.16.0";

    src = fetchFromGitHub {
      owner = "hugopl";
      repo = "gi-crystal";
      rev = "refs/tags/v${version}";
```

Reply:
```
error: undefined variable 'version'

       at /home/sunder/repos/nixpkgs/pkgs/applications/misc/collision/default.nix:20:27:

           19|       repo = "gi-crystal";
           20|       rev = "refs/tags/v${version}";
             |                           ^
           21|       hash = "sha256-ij4U8BoSNHpkh5SmvoH5anGB/ZdvmC5zDwJCusR/s/c=";
 ```
 maybe if move this into gi-crystal/default.nix, it might work.

Reply:
Yes it works